sdk/keyvault/azure-keyvault-keys/azure/keyvault/keys/models.py

```diff
@@ -183,21 +184,72 @@ def tags(self):
 
 
 class Key(KeyBase):
-    """A key's attributes and cryptographic material"""
+    """
+    A key's management attributes and cryptographic material.
+
+    :param str key_id:
+        Key Vault's identifier for the key. Typically a URI, e.g. https://myvault.vault.azure.net/keys/my-key
+    :param jwk:
+        The key's cryptographic material as a JSON Web Key (https://tools.ietf.org/html/rfc7517). This may be provided
+        as a dictionary, keyword arguments, or an instance of :class:`~JsonWebKey`.
+
+    Providing cryptographic material as keyword arguments:
+
+    .. code-block:: python
+
+        from azure.keyvault.keys.models import Key
+
+        key_id = 'https://myvault.vault.azure.net/keys/my-key'
+        key_bytes = os.urandom(32)
+        key = Key(key_id, k=key_bytes, kty='oct', key_ops=['unwrapKey', 'wrapKey'])
+
+    Providing cryptographic material as a dictionary:
+
+    .. code-block:: python
+
+        from azure.keyvault.keys.models import Key
+
+        key_id = 'https://myvault.vault.azure.net/keys/my-key'
+        key_bytes = os.urandom(32)
+        jwk = {'k': key_bytes, 'kty': 'oct', 'key_ops': ['unwrapKey', 'wrapKey']}
+        key = Key(key_id, jwk=jwk)
+
+    """
 
-    def __init__(self, attributes, vault_id, key_material, **kwargs):
-        # type: (_models.KeyAttributes, str, _models.JsonWebKey, **Any) -> None
-        super(Key, self).__init__(attributes, vault_id, **kwargs)
-        self._key_material = key_material
+    def __init__(self, key_id, jwk=None, **kwargs):
+        # type: (str, Optional[dict], **Any) -> None
+        super(Key, self).__init__(key_id=key_id, **kwargs)
+        if isinstance(jwk, dict):
+            self._key_material = JsonWebKey(**jwk)
+        else:
+            self._key_material = JsonWebKey(**kwargs)
```

Comment:
supposed to be *either* jwk *or* kwargs provided, but not both? Would suggest raising a ValueError if both are given

Reply:
Do you think the choice is a strange one to offer? Would kwargs alone be more idiomatic, and callers can just splat a dict if that's what they have?

Reply:
I think kwargs alone would be fine, for sure. I do think allowing `jwk` dict as a convenience is also not unreasonable as long as you check to make sure you only get one or the other.